src/Mvc/src/Microsoft.AspNetCore.Mvc.ViewFeatures/Infrastructure/UnsupportedJavaScriptRuntime.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures
+{
+    internal class UnsupportedJavaScriptRuntime : IJSRuntime
+        {
+            public Task<T> InvokeAsync<T>(string identifier, params object[] args)
+            {
+                throw new InvalidOperationException("JavaScript interoperability is not supported in the current environment.");
```

Comment:
We throw because there's no better thing to do and this way at least you get a chance to see the issue early (hopefully during development).

Writing functional in memory tests for prerrendering should be pretty trivial with WebApplicationFactory and MVC.

Reply:
For extra clarity, could the message say something like:

```
JavaScript interop calls cannot be issued during server-side prerendering, because the page has not yet loaded in the browser. Prerendered components must wrap any JavaScript interop calls in conditional logic to ensure those interop calls are not attempted during prerendering.
```

Maybe rename the type to `PrerenderingUnsupportedJavaScriptRuntime` if that helps with having such a specific error message.